### DISCUSSION TITLE
Not detecting devices in other VLAN (Unifi)

### DISCUSSION THREAD

**Discussion by bsdev90** [netalertx/NetAlertX#649] [Q&A]
``Hi,
I installed NetAlertX via docker on an Unraid server (IP 10.0.1.9) and set the network type to host (otherwise it does not see anything).
The problem is that it only sees the devices of the 10.0.1.0 VLAN.

Here is my Unifi config:
![Capture d'écran 2024-04-23 101622](https://github.com/jokob-sk/NetAlertX/assets/7091140/af776cda-6170-4272-a4e0-34647c734bac)

Here is my NetAlertX config:
![Capture d'écran 2024-04-23 101852](https://github.com/jokob-sk/NetAlertX/assets/7091140/8b75612b-c00a-40b2-886c-ccbdb8af78a6)

I chose br0 because it seems to be the good interface:
![Capture d'écran 2024-04-23 101750](https://github.com/jokob-sk/NetAlertX/assets/7091140/c324a12d-4173-4685-8fa1-89a667654d99)

I tested to do some arp-scan via the container.
The same VLAN works:
```
#  arp-scan 10.0.1.0/24 --interface=br0
Interface: br0, type: EN10MB, MAC: 48:21:0b:36:e4:2e, IPv4: 10.0.1.9
Starting arp-scan 1.10.0 with 256 hosts (https://github.com/royhills/arp-scan)
10.0.1.1        e2:63:da:20:0a:da       (Unknown: locally administered)
10.0.1.2        90:09:d0:32:07:bd       Synology Incorporated
10.0.1.6        52:54:00:4f:1f:48       QEMU
10.0.1.126      52:54:00:15:08:90       QEMU

24 packets received by filter, 0 packets dropped by kernel
Ending arp-scan 1.10.0: 256 hosts scanned in 1.861 seconds (137.56 hosts/sec). 4 responded
```

But the other don't:
```
#  arp-scan 10.0.2.0/24 --interface=br0
Interface: br0, type: EN10MB, MAC: 48:21:0b:36:e4:2e, IPv4: 10.0.1.9
Starting arp-scan 1.10.0 with 256 hosts (https://github.com/royhills/arp-scan)

218 packets received by filter, 0 packets dropped by kernel
Ending arp-scan 1.10.0: 256 hosts scanned in 1.879 seconds (136.24 hosts/sec). 0 responded
```

And tested with the `vlan` parameter, same thing:
```
#  arp-scan 10.0.2.0/24 --interface=br0 --vlan=3
Interface: br0, type: EN10MB, MAC: 48:21:0b:36:e4:2e, IPv4: 10.0.1.9
Starting arp-scan 1.10.0 with 256 hosts (https://github.com/royhills/arp-scan)

2 packets received by filter, 0 packets dropped by kernel
Ending arp-scan 1.10.0: 256 hosts scanned in 1.882 seconds (136.03 hosts/sec). 0 responded
```

I'm pretty sure it's a Unifi problem but I post here because if someone already had this problem and solved it, or if someone has a solution...

Thank !

**Comment by jokob-sk** [comment]
Hi,

I don't have any segmentation, but I run Unify myself. You could try other scan methods, there is also a Unifi import option:

https://github.com/jokob-sk/NetAlertX/tree/main/front/plugins/unifi_import

I also added an NMAP dev discovery plugin that can be tested in the `netalertx-dev` image. It might work better than ARP-SCAN, but unsure. 

Maybe someone else from the community can help as well.
